heudiconv/convert.py

```diff
@@ -851,6 +851,57 @@ def nipype_convert(
     return eg, prov_file
 
 
+def filter_partial_volumes(
+    nii_files: list[str],
+    bids_files: list[str],
+    bids_metas: list[dict[str, Any]],
+):
+    """filter interrupted 4D scans volumes with missing slices on XA: see dcm2niix #742
+
+    Parameters
+    ----------
+    nii_files : list[str]
+        converted nifti filepaths
+    bids_files: list[str]
+        converted BIDS json filepaths
+    bids_metas : list[dict[str, Any]]
+        list of metadata dict loaded from BIDS json files
+
+    Returns
+    -------
+    nii_files
+        filtered niftis
+    bids_files
+        filtered BIDS jsons
+    bids_metas
+        filtered BIDS metadata
+
+    """
+    partial_volumes = [not metadata.get("RawImage", True) for metadata in bids_metas]
```

Comment:
so, a partial volume is iff `RawImage = False`? i.e. is every volume with `RawImage = False` is a partial volume?

Reply:
FTR, `datalad foreach-dataset -r --output-streams relpath   git grep RawImage` across our datasets found  251 of .json files among which
- 241 of _rec-moco files (motion corrected on the scanner I assume).  **So we should not just skip those AFAIK!**
- other 10 are mix of T1w and 1 dwi without having __dup in name with some __dup (very likely to be the canceled ones)

surprised that we didn't see that many of __dup files which should be the interrupted runs.  But likely because conversion was done with earlier versions of dcm2niix so it didn't detected as many cases before